tensorflow/python/keras/engine/compile_utils.py

```diff
@@ -415,9 +419,22 @@ def update_state(self, y_true, y_pred, sample_weight=None):
   def _get_metric_objects(self, metrics, y_t, y_p):
     """Convert user-supplied metrics to `Metric` objects."""
     metrics = nest.flatten(metrics)
-    return [self._get_metric_object(m, y_t, y_p) for m in metrics]
-
-  def _get_metric_object(self, metric, y_t, y_p):
+    if self._loss_container:
+      losses = self._loss_container._losses
```

Comment:
@omalleyt12 If all these are required can we extrapolate in a public function in `LossesContainer` outside of `build`. Or is it safe to call the full `build` method directly?

Reply:
@omalleyt12 Can you please assist on above comments from @bhack. Thanks!

Reply:
@omalleyt12, Any update on this PR? Please. Thanks!